src/core/event/poll_group.h

```diff
@@ -57,8 +58,7 @@ class poll_group {
     void add_dirty_socket(sockinfo_tcp *si);
     void flush();
 
-    void add_ring(ring *);
-    void del_ring(ring *);
```

Comment:
btw, when scket is deleted and it's ring is no longer needed in the group, it still will remain in the group....

Reply:
Yes, this is intentional to avoid ring thrashing in the scenario when application removes all the sockets and then creates new ones frequently. Ring destruction is too expensive now.

Also one of the main reasons to keep rings is efficient RX buffer release which can be performed after the socket destruction. Without keeping the ring, we would need to search for the ring in the group collection and return the buffer to the global buffer pool in corner cases. This would have a cost for the extra lookup.

Reply:
Looks like an important comment to add :)

Reply:
There is already a comment in the `poll_group::add_ring()` implementation